core/node/crypto/chain_monitor.go

```diff
@@ -314,10 +340,10 @@ func (ecm *chainMonitor) RunWithBlockPeriod(
 			}
 
 			callbacksExecuted.Wait()
-			ecm.muBuilder.Unlock()
 
 			// from and toBlocks are inclusive, start at the next block on next iteration
-			fromBlock = new(big.Int).Add(query.ToBlock, one)
+			ecm.setFromBlockVersioned(fromBlockVersion, new(big.Int).Add(query.ToBlock, one))
```

Comment:
why versioned updates here? Do we expect this method to be entered by multiple threads? And also arent updates already write protected by the mutex?